src/extractor/guidance/turn_discovery.cpp

```diff
@@ -55,6 +55,13 @@ bool findPreviousIntersection(const NodeID node_v,
     // (looking at the reverse direction).
     const auto node_w = node_based_graph.GetTarget(via_edge);
     const auto u_turn_at_node_w = intersection[0].eid;
+    BOOST_ASSERT(node_based_graph.BeginEdges(node_w) <= u_turn_at_node_w &&
+                 u_turn_at_node_w <= node_based_graph.EndEdges(node_w));
```

Comment:
different directions. one is from the node prior to the intersection (node_v to node_w) and the uturn is the other way around (node_w to node_v)